lib/puppet/provider/scheduled_task/taskscheduler_api2.rb

```diff
@@ -143,9 +142,9 @@ def trigger=(value)
   end
 
   def user=(value)
-    self.fail("Invalid user: #{value}") unless Puppet::Util::Windows::SID.name_to_sid(value)
+    raise("Invalid user: #{value}") unless Puppet::Util::Windows::SID.name_to_sid(value)
 
-    if value.to_s.downcase != 'system'
+    if !value.to_s.casecmp('system').zero?
```

Comment:
Not a blocker, but, add this to my growing list of rubocop suggestions that I believe are more obtuse than the original code.